gateway-ha/src/test/java/com/lyft/data/gateway/ha/router/TestPrestoQueueLengthRoutingTable.java

```diff
@@ -197,6 +236,70 @@ public void testRoutingWithSkewedWeightDistribution() {
     }
   }
 
+  @Test
+  public void testRoutingWithUserQueuedLength() {
+    int numBackends = 2;
+    int queryVolume = 10000;
+
+    // Case 1: All user queue counts Present.
+    // Validate always routed to  cluster with lowest user queue
+    registerBackEndsWithUserQueue(mockRoutingGroup, numBackends, Arrays.asList(1, 2));
+    for (int i = 0; i < queryVolume; i++) {
+      assertEquals(routingTable.getEligibleBackEnd(mockRoutingGroup, mockUser),
+              mockRoutingGroup + "0");
+    }
+
+    // Case 2: Not all user queue counts Present.
+    // Validate always routed to cluster with zero queue length i.e. the missing cluster.
+    registerBackEndsWithUserQueue(mockRoutingGroup, numBackends, Arrays.asList(1));
+    for (int i = 0; i < queryVolume; i++) {
+      assertEquals(routingTable.getEligibleBackEnd(mockRoutingGroup, mockUser),
+              mockRoutingGroup + "1");
+    }
+
+    // Case 3: All user queue counts Present but equal
```

Comment:
Just to confirm, is the intention for all of these 3 scenarios fall back to the old weighted logic? 
Scenario 3) In `getEligibleBackEnd`, the `leastQueuedCluster` is abandoned if `leastQueuedCluster` = `mostQueuedCluster` 
Scenario 4) new logic is not applied if user queue counts don't exist
Scenario 5) new logic is not applied if user does not exist

Reply:
Yup, we fallback for all 3 scenarios as there is no decision which can be made using the user level queue counts in these cases.